modules/pam_timestamp/hmac_openssl_wrapper.c

```diff
@@ -81,22 +86,39 @@ get_crypto_algorithm(pam_handle_t *pamh, int debug){
 }
 
 static int
+PAM_NONNULL((1, 2))
 generate_key(pam_handle_t *pamh, char **key, size_t key_size)
 {
     int fd = 0;
     ssize_t bytes_read = 0;
-    char * tmp = NULL;
+    char *tmp = *key = NULL;
 
-    fd = open("/dev/urandom", O_RDONLY);
-    if (fd == -1) {
-        pam_syslog(pamh, LOG_ERR, "Cannot open /dev/urandom: %m");
+    tmp = calloc(key_size, 1);
```

Comment:
What was the reason of replacing `malloc` with `calloc`?

Reply:
Having consistent, zero-initialized memory. This ensured a deterministic state when later operations to fill this memory only return the memory only partially filled.

No deal-breaker if skipped, but preferable IMO.